cmd/boots/main_test.go

```diff
@@ -45,6 +46,90 @@ func TestParser(t *testing.T) {
 	}
 }
 
+func TestParseDynamicIPXEVarsFunc(t *testing.T) {
+	tests := []struct {
+		name     string
+		ipxevars string
+		want     [][]string
+		wantErr  bool
+	}{
+		{
+			name:     "Empty string input",
+			ipxevars: "",
+			want:     nil,
+			wantErr:  false,
+		},
+		{
+			name:     "Single var definition",
+			ipxevars: "myvar1=myval1",
+			want:     [][]string{{"myvar1", "myval1"}},
+			wantErr:  false,
+		},
+		{
+			name:     "Two var definitions",
+			ipxevars: "myvar1=myval1 myvar2=myval2",
+			want:     [][]string{{"myvar1", "myval1"}, {"myvar2", "myval2"}},
+			wantErr:  false,
+		},
+		{
+			name:     "Single quotes in var definition",
+			ipxevars: "'myvar1'='myval1'",
+			want:     [][]string{{"'myvar1'", "'myval1'"}},
+			wantErr:  false,
+		},
+		{
+			name:     "Double quotes in var definition",
+			ipxevars: "\"myvar1\"=\"myval1\"",
+			want:     [][]string{{"\"myvar1\"", "\"myval1\""}},
+			wantErr:  false,
+		},
+		{
+			name:     "Invalid var definition - no equals specified",
+			ipxevars: "abcdefg",
+			want:     nil,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid var definition - spaces inside varname",
+			ipxevars: "my var one=myval1",
+			want:     nil,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid var definition - spaces inside value",
+			ipxevars: "myvar1=my val one",
+			want:     nil,
+			wantErr:  true,
+		},
+		{
+			name:     "Invalid var definition - just passing '='",
+			ipxevars: "=",
+			want:     nil,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := parseDynamicIPXEVars(tt.ipxevars)
+			if tt.wantErr {
+				if err != nil {
+					// pass
+					return
+				}
+				t.Fatalf("parseDynamicIPXEVars() did not return an error, instead returned %v", got)
+			}
+			if err != nil {
+				t.Fatalf("parseDynamicIPXEVars() returned an unexpected error: %s", err)
+			}
+
+			want := tt.want
+			if !reflect.DeepEqual(want, got) {
```

Comment:
You might want to use `cmp.Diff` or `cmp.Equal` instead of `reflect.DeepEqual`.

_"You will find older code using the standard reflect.DeepEqual function to compare complex structures. Prefer cmp for new code, and consider updating older code to use cmp where practical. reflect.DeepEqual is sensitive to changes in unexported fields and other implementation details."_ -- [ref](https://github.com/golang/go/wiki/TestComments#equality-comparison-and-diffs)

Reply:
Cool, I've now switched this to using cmp.Equal.